### PR TITLE
Scripts to bulk edit user's email addresses and user group names from CSV

### PR DESCRIPTION
**Pull request by nichollsdave**
The use case is that the domain for emails are changing along with user groups due to the Active Directory they are authenticating against changing.  Given a list of old email addresses to new and old user group names to new the scripts will update the users/groups accordingly.

### CODE REVIEW

File: examples/update_usergroups_from_csv.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        logging.info(f"{failed} user groups failed to update")
+
+def update_user_group(hub_client, existing_name, new_name, user_group):
+    user_group_url = hub_client.list_resources(user_group)['href']
+
+    # Update the name.
+    user_group['name'] = new_name
+
+    logging.info(f"Updating user group {existing_name} to {user_group['name']} for user group {user_group_url}s")
+    hub_client.session.put(user_group_url, json=user_group)
```
Comment (koshmack) [inline examples/update_usergroups_from_csv.py:77]: The last character "s" after {user_group_url} is not needed. If "s" is attached to the URL string, that URL probably will not navigate you to the usergroup.

File: examples/update_users_from_csv.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
+'''
```
Comment (koshmack) [inline examples/update_users_from_csv.py:1]: Since this code is almost identical to the usergroups, I have the exact same comments as for the usergroups.

**Review by koshmack — COMMENTED** [review]
Hi David, I have left several comments which are mostly cosmetic ones but you might be interested in the one for Copyright and the other for the odd "s" character in the user or usergroup URL. Koishi

File: examples/update_usergroups_from_csv.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import argparse
+import sys
+import json
+import traceback
+from requests import HTTPError, RequestException
+
+from blackduck import Client
+
+def log_config():
+    # TODO: debug option in .restconfig file to be reflected
```
[the remark this answers is not in the document]
Comment (nichollsdave) [reply examples/update_usergroups_from_csv.py]: Good catch I have removed it

File: examples/update_usergroups_from_csv.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+    logging.info(f"Updating user group {existing_name} to {user_group['name']} for user group {user_group_url}s")
+    hub_client.session.put(user_group_url, json=user_group)
+
+
+def main():
+    log_config()
+    args = parse_parameter()
+    try:
+        with open('.restconfig.json','r') as f:
```
[the remark this answers is not in the document]
Comment (nichollsdave) [reply examples/update_usergroups_from_csv.py:114]: I have been meaning to write an article on this and have done so and pointed to it in the comment as well as giving the basics.  Good to finally get round to it.

File: examples/update_usergroups_from_csv.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
+'''
+Created on Jan 27, 2023
+@author: dnichol
+Bulk update user group names from CSV.  CSV file requires two columns, titled 'Existing' and 'New'.  This script is case sensitive.
+'''
+
```
[the remark this answers is not in the document]
Comment (nichollsdave) [reply examples/update_usergroups_from_csv.py:37]: Can't believe I forgot my own guidance on this :)

File: examples/update_usergroups_from_csv.py (hunk 1 above)
Comment (nichollsdave) [reply examples/update_usergroups_from_csv.py:107]: Wow you have great eyes, I missed that.

File: examples/update_users_from_csv.py (hunk 2 above)
Comment (nichollsdave) [reply examples/update_users_from_csv.py:1]: Applied changes to both files.

**Review by koshmack — APPROVED** [review]
Thanks for updating the code. No more comments and approved. Great works!

**Pull request merged**